src/main.rs

```
@@ -87,37 +92,72 @@ async fn config_cmd(file: Option<String>) {
             };
 
             let Ok(ip) = res else {
-                continue;
+                return Err("Failed to get public ip".to_string());
             };
 
-            println!("IP found for domain {} => {}", &domain.name, &ip);
-            let res = duckdns
+            let domains = vec![domain.name.clone()];
+            if let Err(_) = duckdns
                 .update(
```

Using `if let Err(_)` without handling the error can lead to silent failures. It's better to log or handle the error appropriately to avoid unexpected behavior.